iron-spi-aws/src/main/java/io/axway/iron/spi/aws/s3/AwsS3Utils.java

```diff
@@ -0,0 +1,160 @@
+package io.axway.iron.spi.aws.s3;
+
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.CreateBucketRequest;
+import com.amazonaws.services.s3.model.HeadBucketRequest;
+import com.amazonaws.services.s3.model.ListVersionsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.S3VersionSummary;
+import com.amazonaws.services.s3.model.VersionListing;
+
+import static io.axway.iron.spi.aws.AwsProperties.*;
+import static io.axway.iron.spi.aws.PropertiesHelper.getValue;
+import static io.axway.iron.spi.aws.s3.AwsS3Properties.*;
+
+public class AwsS3Utils {
+    private static final Logger LOG = LoggerFactory.getLogger(AwsS3Utils.class);
+
+    /**
+     * Create a AwsS3SnapshotStoreFactory with some properties set to configure S3 client:
+     * - aws access key (optional+) {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_ACCESS_KEY_PROPERTY} / {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_ACCESS_KEY_ENVVAR}
+     * - aws secret key (optional+) {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_SECRET_KEY_PROPERTY} / {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_SECRET_KEY_ENVVAR}
+     * - aws region (optional*) {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_REGION_PROPERTY} / {@value io.axway.iron.spi.aws.AwsProperties.Constants#AWS_REGION_ENVVAR}
+     * - s3 endpoint (optional*) {@value io.axway.iron.spi.aws.s3.AwsS3Properties.Constants#AWS_S3_ENDPOINT_PROPERTY} / {@value io.axway.iron.spi.aws.s3.AwsS3Properties.Constants#AWS_S3_ENDPOINT_ENVVAR}
+     * - s3 port (optional*) {@value io.axway.iron.spi.aws.s3.AwsS3Properties.Constants#AWS_S3_PORT_PROPERTY} / {@value io.axway.iron.spi.aws.s3.AwsS3Properties.Constants#AWS_S3_PORT_ENVVAR}
+     * (+) to configure the access, both access key and secret key must be provided.
+     * (*) to configure the endpoint URL, the endpoint, the port and the region must be provided.
+     *
+     * @param properties the properties
+     */
+    public static AmazonS3 buildS3Client(Properties properties) {
```

Comment:
look likes some of that method could be put in common with the kinesis spi

Reply:
Done.